### PR TITLE
YMKM 10 cards

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/perforator_crocodile.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Name:Perforator Crocodile
+ManaCost:3 B B
+Types:Creature Crocodile
+PT:6/5
+K:Cycling:1 B
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigRepeat | TriggerDescription$ When CARDNAME enters the battlefield, for each creature your opponents control, conjure a card named Stab Wound onto the battlefield attached to that creature.
+SVar:TrigRepeat:DB$ RepeatEach | RepeatCards$ Creature.OppCtrl | RepeatSubAbility$ DBConjure
```
Comment (Simisays) [inline forge-gui/res/cardsfolder/upcoming/perforator_crocodile.txt:7]: This causes a crash 
image

Any thoughts why @Northmoc

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/perforator_crocodile.txt:7]: the LastState params in MakeCard should be obtained from Game

File: forge-gui/res/cardsfolder/upcoming/case_of_the_lost_witness.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+Name:Case of the Lost Witness
+ManaCost:1 U
+Types:Enchantment Case
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigConjure | TriggerDescription$ When this Case enters the battlefield, conjure four cards named Fblthp, the Lost into your library, then shuffle. Draw a card.
+SVar:TrigConjure:DB$ MakeCard | Conjure$ True | Name$ Fblthp, the Lost | Zone$ Library | Amount$ 4 | SubAbility$ DBDraw
+SVar:DBDraw:DB$ Draw
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Battlefield | IsPresent$ Homunculus.YouCtrl+Legendary | IsPresent2$ Card.Self+IsUnsolved | Execute$ TrigSolve | TriggerDescription$ To solve — You control a legendary Homunculus.
+SVar:TrigSolve:DB$ AlterAttribute | Defined$ Self | Attributes$ Solved
+Solved — You may look at the top card of your library any time and you may play lands and cast spells from the top of your library.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/case_of_the_lost_witness.txt:9]: bad line

File: forge-gui/res/cardsfolder/upcoming/emmara_voice_of_the_conclave.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+Name:Emmara, Voice of the Conclave
+ManaCost:1 G W
+Types:Legendary Creature Elf Cleric
+PT:2/3
+K:Convoke
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigDraft | TriggerDescription$ When CARDNAME enters the battlefield, draft a card from CARDNAME's spellbook.When CARDNAME enters the battlefield, draft a card from CARDNAME's spellbook.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/emmara_voice_of_the_conclave.txt:6]: fix description

File: forge-gui/res/cardsfolder/upcoming/spotlight_falcon.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+Name:Spotlight Falcon
+ManaCost:2 W W
+Types:Artifact Creature Bird
+PT:4/3
+S:Mode$ OptionalCost | EffectZone$ All | ValidCard$ Card.Self | ValidSA$ Spell | Cost$ CollectEvidence<5> | Description$ As an additional cost to cast this spell, you may collect evidence 5.
+K:Flying
+T:Mode$ ChangesZone | Origin$ Any | CheckSVar$ X | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigConjure | TriggerDescription$ When this Case enters the battlefield, conjure four cards named Fblthp, the Lost into your library, then shuffle. Draw a card.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/spotlight_falcon.txt:7]: fix description

File: forge-gui/res/cardsfolder/upcoming/cackling_observer.txt [hunk 5]
```diff
@@ -0,0 +1,14 @@
+Name:Cackling Observer
+ManaCost:1 B B
+Types:Creature Imp
+PT:3/1
+K:Flying
+T:Mode$ ChangesZone | ValidCard$ Card.Self | Origin$ Any | Destination$ Battlefield | Execute$ TrigReveal | TriggerDescription$ When CARDNAME enters the battlefield, target opponent reveals each nonland card in their hand. You choose one of those cards and exile it.
+SVar:TrigReveal:DB$ Reveal | ValidTgts$ Opponent | RevealAllValid$ Card.nonLand+TargetedPlayerCtrl | RememberRevealed$ True | SubAbility$ DBExile
+SVar:DBExile:DB$ ChangeZone | Origin$ Hand | Destination$ Exile | DefinedPlayer$ Targeted | Chooser$ You | ChangeType$ Card.nonLand+nonCreature | Imprint$ True | Mandatory$ True  | SubAbility$ DBClearRemembered
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/cackling_observer.txt:8]: spacing `Mandatory$ True  | `

File: forge-gui/res/cardsfolder/upcoming/case_of_the_lost_witness.txt [hunk 6]
```diff
@@ -0,0 +1,11 @@
+Name:Case of the Lost Witness
+ManaCost:1 U
+Types:Enchantment Case
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigConjure | TriggerDescription$ When this Case enters the battlefield, conjure four cards named Fblthp, the Lost into your library, then shuffle. Draw a card.
+SVar:TrigConjure:DB$ MakeCard | Conjure$ True | Name$ Fblthp, the Lost | Zone$ Library | Amount$ 4 | SubAbility$ DBDraw
+SVar:DBDraw:DB$ Draw
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Battlefield | IsPresent$ Homunculus.YouCtrl+Legendary | IsPresent2$ Card.Self+IsUnsolved | Execute$ TrigSolve | TriggerDescription$ To solve — You control a legendary Homunculus.
+SVar:TrigSolve:DB$ AlterAttribute | Defined$ Self | Attributes$ Solved
+S:Mode$ Continuous | Affected$ Land.TopLibrary+YouCtrl,Card.TopLibrary+YouCtrl | IsPresent$ Card.Self+IsSolved | AffectedZone$ Library | MayLookAt$ You | MayPlay$ True | Description$ Solved — You may look at the top card of your library any time and you may play lands and cast spells from the top of your library.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/case_of_the_lost_witness.txt:9]: does this not work just with `Affected$ Card.TopLibrary+YouCtrl` ?

File: forge-gui/res/cardsfolder/upcoming/excogitator_sphinx.txt [hunk 7]
```diff
@@ -0,0 +1,11 @@
+Name:Excogitator Sphinx
+ManaCost:2 U U
+Types:Creature Sphinx Detective
+PT:4/4
+K:Flying
+T:Mode$ DamageDoneOnce | ValidSource$ Creature.YouCtrl | TriggerZones$ Battlefield | ValidTarget$ Player | CombatDamage$ True | Execute$ TrigInvestigate | TriggerDescription$ Whenever one or more creatures you control deal combat damage to a player, investigate.
+SVar:TrigInvestigate:DB$ Investigate
+A:AB$ Seek | Cost$ 1 Sac<1/Clue> | Type$ Instant,Sorcery | SpellDescription$ Seek an instant or sorcery card.
+DeckHas:Ability$Investigate|Token|Sacrifice & Type$Artifact|Clue
+DeckHints:Type$Clue
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/excogitator_sphinx.txt:10]: |Sorcery|Instant

File: forge-gui/res/cardsfolder/upcoming/woodland_investigation.txt [hunk 8]
```diff
@@ -0,0 +1,9 @@
+Name:Woodland Investigation
+ManaCost:G
+Types:Sorcery
+A:SP$ ChangeZone | Origin$ Library | Destination$ Hand | ChangeType$ Land.Basic | ChangeNum$ 1 | RememberChanged$ True | SubAbility$ DBAnimate | SpellDescription$ Search your library for a basic land card, reveal it, put it into your hand, then shuffle. 
+SVar:DBAnimate:DB$ Animate | Types$ Clue,Artifact | Duration$ Perpetual | Abilities$ ClueSac | Defined$ Remembered | SubAbility$ DBCleanup | SpellDescription$ It perpetually becomes a Clue artifact in addition to its other types and gains "{2}, Sacrifice this permanent: Draw a card."
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+SVar:ClueSac:AB$ Draw | Cost$ 2 Sac<1/CARDNAME/this artifact> | NumCards$ 1 | SpellDescription$ Draw a card.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/woodland_investigation.txt:7]: this permanent

File: forge-gui/res/cardsfolder/upcoming/excogitator_sphinx.txt [hunk 9]
```diff
@@ -7,5 +7,5 @@ T:Mode$ DamageDoneOnce | ValidSource$ Creature.YouCtrl | TriggerZones$ Battlefie
 SVar:TrigInvestigate:DB$ Investigate
 A:AB$ Seek | Cost$ 1 Sac<1/Clue> | Type$ Instant,Sorcery | SpellDescription$ Seek an instant or sorcery card.
 DeckHas:Ability$Investigate|Token|Sacrifice & Type$Artifact|Clue
-DeckHints:Type$Clue
+DeckHints:Type$Clue|Sorcery|Instant
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/excogitator_sphinx.txt:10]: Investigate makes a clue, right? So we should have some DeckHas too?

Maybe upgrade sorcery/instant to DeckNeeds? Is this card any good if that whiffs?

File: forge-gui/res/cardsfolder/upcoming/mapping_the_maze.txt [hunk 10]
```diff
@@ -0,0 +1,9 @@
+Name:Mapping the Maze
+ManaCost:1 U
+Types:Instant
+A:SP$ ChooseCard | ChoiceZone$ Hand,Graveyard | Choices$ Instant.YouOwn,Sorcery.YouOwn | ChoiceTitle$ Choose an instant or sorcery card in your hand or graveyard | SubAbility$ DBAnimate | SpellDescription$ Choose an instant or sorcery card in your hand or graveyard. 
+SVar:DBAnimate:DB$ Animate | Triggers$ SpellCastTrig | Defined$ ChosenCard | Incorporate$ U | Duration$ Perpetual | SubAbility$ DBReturn | SpellDescription$ That card perpetually incorporates {U} and gains "When you cast this spell, draw a card."
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/mapping_the_maze.txt:5]: this probably needs StackDescription of some sort?

File: forge-gui/res/cardsfolder/upcoming/mapping_the_maze.txt [hunk 11]
```diff
@@ -0,0 +1,9 @@
...
+SVar:DBReturn:DB$ ChangeZone | Origin$ Graveyard | Destination$ Hand | ChangeTypeDesc$ a multicolored instant or sorcery card from your graveyard | ChangeNum$ 1 | Hidden$ True | ChangeType$ Instant.YouCtrl+MultiColor,Sorcery.YouCtrl+MultiColor | SpellDescription$ Then return a multicolored instant or sorcery card from your graveyard to your hand.
+SVar:SpellCastTrig:Mode$ SpellCast | ValidCard$ Card.Self | Execute$ TrigDraw | TriggerDescription$ When you cast this spell, draw a card.
+SVar:TrigDraw:DB$ Draw
+Oracle:Choose an instant or sorcery card in your hand or graveyard. That card perpetually incorporates {U} and gains "When you cast this spell, draw a card." Then return a multicolored instant or sorcery card from your graveyard to your hand.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/mapping_the_maze.txt:9]: DeckNeeds Instant Sorcery, right?
DeckHas Graveyard
I don't know if there's a good AI flag that this should be in a multicolor deck
If not we may RemoveDeck:Random ?

File: forge-gui/res/cardsfolder/upcoming/scrutiny_of_the_guildpact.txt [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+ManaCost:1 W
+Types:Enchantment
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigChoose | TriggerDescription$ When CARDNAME enters the battlefield, you may choose a creature card in your hand. If you do, that card perpetually incorporates {W} and gains "When you cast this spell, create a 2/2 white and blue Detective creature token."
+SVar:TrigChoose:DB$ ChooseCard | ChoiceZone$ Hand | Choices$ Creature.YouOwn | ChoiceTitle$ You may choose a creature card in your hand | SubAbility$ DBAnimate
+SVar:DBAnimate:DB$ Animate | Defined$ ChosenCard | Incorporate$ W | Triggers$ SpellCastTrig | Duration$ Perpetual
+SVar:SpellCastTrig:Mode$ SpellCast | ValidCard$ Card.Self | Execute$ TrigToken | TriggerDescription$ When you cast this spell, create a 2/2 white and blue Detective creature token.
+SVar:TrigToken:DB$ Token | TokenScript$ wu_2_2_detective
+S:Mode$ Continuous | Affected$ Creature.MultiColor+Other+YouCtrl | AddPower$ 1 | AddToughness$ 1 | Description$ Multicolored creatures you control get +1/+1.
+DeckHints:Type$Detective
+DeckHas:Ability$Token & Type$Detective
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/scrutiny_of_the_guildpact.txt:11]: Color$Blue

File: forge-gui/res/cardsfolder/upcoming/mapping_the_maze.txt (hunk 10 above)
Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/mapping_the_maze.txt:5]: I don't think this is quite resolved

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/mapping_the_maze.txt:5]: Sorry resolved this too quickly. I think this way the stackdescription is fine ?

**Review by Northmoc — APPROVED** [review]
Looks good to me! 👍

**Pull request merged**